### PR TITLE
feat: Terminal asks for student name and adds to prompt.

### LINKED ISSUE #31: Custom Bash prompt with Student Name
The first time a Terminal runs in KitClient it should prompt the student for a name.  This name should then be included in the Bash prompt.  This ensures that the student is identified in each of the screenshots that they take for the activities.  It is a simple mechanism that makes it difficult for a student to not do their own work.

Ideally the prompt will include both the current directory and the student's name.

### PR DESCRIPTION
**Pull request by braughtg**
Closes #31 

This PR will cause the Terminal (`bash`) to prompt the student for their name the first time it is opened.  This name will then become part of the command prompt in the terminal. 

To Verify:
1. remove prior container/image and volume
2. create container as per README (append `:pr-36` to image name)
3. start container as per README
4. connect via vnc or novnc
5. open a terminal
6. should be prompted for name - enter a name
7. prompt should change to include name and pwd.
8. open new terminal - prompt should be same as in 7.
9. stop container
10. restart container 
11. open terminal - prompt should be the same as in 7.
12. stop container as per README
13. delete container as per README
14. create container
15. start container
16. open terminal - prompt should be the same as in 7.

### CODE REVIEW

**Comment by kwurst** [comment]
@braughtg Are you expecting more than just a first name? When I entered first and last name, I got the following:

Please enter your name: Karl Wurst
bash: [: Karl: binary operator expected
Karl Wurst:~ $

**Comment by kwurst** [comment]
> @braughtg Are you expecting more than just a first name? When I entered first and last name, I got the following:
> 
> Please enter your name: Karl Wurst bash: [: Karl: binary operator expected Karl Wurst:~ $

@braughtg Other than that error message it works correctly.

**Comment by braughtg** [comment]
Good catch! I’ll patch that up.

On Jan 13, 2023, at 16:40, Karl R. Wurst ***@***.***> wrote:

﻿
Caution:This email originated from outside of the College. Do not click links or open attachments unless you recognize the sender and know the content is safe. Report anything suspicious to the HelpDesk.


@braughtg<https://nam12.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fbraughtg&data=05%7C01%7Cbraught%40dickinson.edu%7C9efdd40cf84041cc6a6c08daf5aed1ed%7C6232b05576b94c139b88b562ae7db6fb%7C0%7C0%7C638092428446764286%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=v7ywYDCQ2C13F0%2FziDajLH%2BVWxvCYD21mRnTI0DoNY4%3D&reserved=0> Are you expecting more than just a first name? When I entered first and last name, I got the following:

Please enter your name: Karl Wurst bash: [: Karl: binary operator expected Karl Wurst:~ $

@braughtg<https://nam12.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fbraughtg&data=05%7C01%7Cbraught%40dickinson.edu%7C9efdd40cf84041cc6a6c08daf5aed1ed%7C6232b05576b94c139b88b562ae7db6fb%7C0%7C0%7C638092428446764286%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=v7ywYDCQ2C13F0%2FziDajLH%2BVWxvCYD21mRnTI0DoNY4%3D&reserved=0> Other than that error message it works correctly.

**Comment by StoneyJackson** [comment]
@braughtg 

My name is not in pwd as step 7 indicates. Here's what it looks like after steps 1-8.

Screenshot 2023-01-14 at 9 55 33 AM

Everything else works fine (for all steps).

**Comment by kwurst** [comment]
@StoneyJackson I think you misunderstood. The prompt contains both your name and the pwd. I don't think is was supposed to rename the home directory to the student's name.

**Comment by StoneyJackson** [comment]
I've added a template for PRs... it should show up on future PRs... but for now here it is:

---

Use a [conventional commit message](https://www.conventionalcommits.org/) in your squash/merge commit message.
In particular, prefix the subject of the squash/merge commit message with one of the following:

* `fix:` if your PR contains no new functionality.
* `feat:` if your PR contains at least one new feature.
* Add an exclemation point to the end of the prefix (e.g., `fix!:`) if your PR contains a ***breaking change***
  (i.e., a change that is not backwards-compatible).

Also, please use [`Co-authored-by:`](https://docs.github.com/en/pull-requests/committing-changes-to-your-project/creating-and-editing-commits/creating-a-commit-with-multiple-authors)
to give credit to others who helped you with this PR.

---

**Comment by StoneyJackson** [comment]
> 

Ah, yes, I misunderstood. Then it looks great!

**Review by StoneyJackson — APPROVED** [review]

**Comment by StoneyJackson** [comment]
When Kit-TTY asks them to run `bash` to continue, they get asked for their name again. So there is a bit of an interaction that we need to work out.

File: src/bashrc.additions [hunk 1]
```diff
@@ -0,0 +1,8 @@
+if [ ! -f .kitclient/kituser.txt ] ; then
+  while [ -z $KIT_USER_NAME ] ; do
+    read -p "Please enter your name: " KIT_USER_NAME
+  done
+  echo $KIT_USER_NAME > .kitclient/kituser.txt
+fi
+KIT_USER_NAME=$(cat .kitclient/kituser.txt) 
+export PS1="\e[1;35m$KIT_USER_NAME\e[m:\e[1;36m\w\e[m \$ "
```
Comment (StoneyJackson) [inline src/bashrc.additions:8]: Please end files with a newline.

In fact, this is causing an interaction between your the KitClient and the Kit-TTY's git-shim.

Comment (StoneyJackson) [reply src/bashrc.additions:8]: This fixes the problem that I was referring to before.

Comment (StoneyJackson) [reply src/bashrc.additions]: I just added it to this branch. I hope no one minds. After CI completes successfully, I'll merge.

Comment (braughtg) [reply src/bashrc.additions]: @StoneyJackson Did this also address the issue raised by @kwurst above?

> When I entered first and last name, I got the following:
> 
> Please enter your name: Karl Wurst bash: [: Karl: binary operator expected Karl Wurst:~ $

Comment (StoneyJackson) [reply src/bashrc.additions]: @braughtg No. I was only addressing the interaction between the feature installer and this code (both hitting .bashrc). I thought you had already fixed @kwurst issue. Sorry, I missunderstood.

**Pull request merged**